storage-api/src/java/org/apache/hadoop/hive/common/TableName.java

```diff
@@ -86,36 +101,55 @@ public String getTable() {
 
   /**
    * Get the name in db.table format, for use with stuff not yet converted to use the catalog.
+   * Fair warning, that if the db is null, this will return null.tableName
+   * @deprecated use {@link #getNotEmptyDbTable()} instead.
    */
+  // to be @Deprecated
   public String getDbTable() {
     return db + DatabaseName.CAT_DB_TABLE_SEPARATOR + table;
+  }
+
+  /**
+   * Get the name in `db`.`table` escaped format, if db is not empty, otherwise pass only the table name.
+   */
+  public String getEscapedNotEmptyDbTable() {
```

Comment:
I think these escaper things should live in the exec side; aren't they only used during unparsing around there?